contrib/gcppubsub/pkg/dispatcher/dispatcher/reconcile.go

```diff
@@ -312,18 +318,29 @@ func (r *reconciler) receiveMessagesBlocking(ctxWithCancel context.Context, enqu
 	}
 }
 
-func receiveFunc(logger *zap.SugaredLogger, sub pubsubutil.GcpPubSubSubscriptionStatus, defaults provisioners.DispatchDefaults, dispatcher provisioners.Dispatcher) func(context.Context, pubsubutil.PubSubMessage) {
+func receiveFunc(logger *zap.SugaredLogger, sub pubsubutil.GcpPubSubSubscriptionStatus, defaults provisioners.DispatchDefaults, dispatcher provisioners.Dispatcher, rateLimiter *workqueue.RateLimiter) func(context.Context, pubsubutil.PubSubMessage) {
```

Comment:
We really should have a unit test for the exponential backoff. That probably implies we need to allow the call to `time.Sleep` to be sent to our own test code (likely just storing the time in a slice and immediately returning). 

The test should test failures to see that it exponentially backs off up to the maximum (and not above) and deletes from the `rateLimiter` once it succeeds.

Reply:
I thought about this sometime back, and concluded that testing the exponential backoff was more of a rateLimiter responsibility. But you are right, I'll add it here

Reply:
Done the changes. Please take a look. 
I still think testing the backoff times is not part of this UT, but rather the rate limiter, but I added anyway.